### ISSUE TITLE
pbtn clarification

### ISSUE THREAD

**Issue by Effer** [nesbox/tic.computer#2]
Not sure if **pbtn** is working as it should. If I keep pressed a key **pbtn** return true every odd frame.
I was expecting to only receive true again after releasing and pressing again the button.
Try in Win10 64Bit  with 0.0.13 and 0.0.12

**Comment by nesbox** [comment]
**pbtn** returns true if button wasn't pressed last tick
also returns true on every 6 ticks if you pressed it more than 30 ticks
it designed to use it with menus or items select (like in pico8)

**Comment by Effer** [comment]
Could be possible to add a sign to the method to make it works as a single shot trigger? Something like **pbtn(0,1)**?
Is the wiki here editable to add some documentation?

**Comment by nesbox** [comment]
Not sure if we need make api more complicated in this case, so you can use code like that
``` lua
local last=0
function btnp(index)
	return btn(index) and (last&(1<<index)==0)
end
function TIC()
	if btnp(4) then 
		trace("pressed A")
	end
	last=btn()
end
```
PS.
Unfortunately I found ugly bug in TIC integration with Lua and the code above will work only in 0.0.14 :confused:

**Comment by Effer** [comment]
Thanks for the nice snippet. I wrote a similar workaround in Tic Tac Toe demo, but have this in the API would make the code looks cleaner.

**Comment by nesbox** [comment]
Ok, will do something like that
`btnp [ id [ hold period ] ] -> pressed`
it will return true every `period` ticks if button is held for `hold` ticks
by default `btnp(id)` will work like "single shot trigger"

**Comment by Effer** [comment]
That will fit really well in the API and will cover most of common needs.

**Comment by nesbox** [comment]
fixed in 0.0.14
